### DISCUSSION TITLE
Where is SiteExistsAsync ?

### DISCUSSION THREAD

**Discussion by kongmengfei** [pnp/pnpcore#891] [Q&A] [answered]
I am using pnp core sdk to check if a site exist but below method did not exist in real program.

[Checking if a site collection exists](https://pnp.github.io/pnpcore/using-the-sdk/admin-sharepoint-sites.html#checking-if-a-site-collection-exists)

![image](https://user-images.githubusercontent.com/57654821/175237171-8e5da85d-a429-4fe9-9631-a69959164078.png)

![image](https://user-images.githubusercontent.com/57654821/175237289-842c7dff-9d0b-4651-bcea-dfcce9b5183b.png)

Am i doing something wrong?

**Comment by jansenbe** [comment, marked as answer]
@kongmengfei : this method only exists in our nightly builds as it was added after release of version 1.6.0. So you can either use a nightly build or wait for 1.7.0, which we plan to release by end of next week.

**Reply by kongmengfei** [reply]
nightly means preview version, right?  oh, I have not recognized this is still being developed as it has been a lot of years.  this library is excellent, the original csom is hard to use. 
by the way does it provide a way to enable site collection app catalog through C#? i am going to create and provision site using this sdk!!

thanks

**Reply by jansenbe** [reply]
@kongmengfei : Thanks! See https://pnp.github.io/pnpcore/using-the-sdk/admin-sharepoint-apps.html#ensuring-theres-a-tenant-app-catalog for ensuring the tenant app catalog exists, we currently don't support something similar for a site collection app catalog but that's a good feature request to add.

**Reply by jansenbe** [reply]
@kongmengfei : already added support for ensuring a site collection app catalog exists. See https://pnp.github.io/pnpcore/using-the-sdk/admin-sharepoint-apps.html#ensuring-theres-a-site-collection-app-catalog

**Reply by jansenbe** [reply]
will be part of the next nightly build and version 1.7 which we'll release this week.
